crates/core/component/shielded-pool/src/genesis/allocation.rs

```diff
@@ -10,16 +14,38 @@ use serde::{Deserialize, Serialize};
     into = "pb::genesis_content::Allocation"
 )]
 pub struct Allocation {
-    pub amount: Amount,
-    pub denom: String,
+    pub raw_amount: Amount,
+    pub raw_denom: String,
```

Comment:
I changed these to `raw_` to reflect that they might be in units that need adjustment. This is not a great pattern -- we should avoid using `Amount` for anything other than an amount of the base denom -- but it's probably okay here because it's only for the genesis data, which has to be trusted anyways.